cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/CacherUtils.java

```diff
@@ -70,6 +77,48 @@ public void cleanTmpFiles() {
         }
     }
 
+    /**
+     * Clean RHPAM and RHDM nightly artifacts.
+     * If there is no artifacts persisted, nothing will be done.
+     * If there is one newer build date and all of the remaining ones are 3 days older than the most recent one, all
+     * of them will be deleted keeping only the most recent .
+     * run once a day.
+     */
+    @Scheduled(every = "24h", delay = 12, delayUnit = TimeUnit.HOURS)
+    public void cleanOldProductNightlyArtifacts() throws IOException, InterruptedException {
+        log.fine("Trying to identify the latest nightly build based on filesystem artifacts to delete, this could take a while...");
+        try {
+            log.fine("Walking through data dir searching for old nightly product builds...");
+
+            Map<File, LocalDate> nightlyBuildArtifacts = new HashMap<>();
+            Files.walk(Paths.get(cacherProperties.getCacherArtifactsDir()))
+                    .map(Path::toFile)
+                    .filter(file -> cacherProperties.buildDatePattern.matcher(file.getName()).find())
+                    .filter(file -> file.getName().endsWith(".zip"))
+                    .forEach(file -> {
+                        Matcher test = cacherProperties.buildDatePattern.matcher(file.getName());
+                        if (test.find()) {
+                            LocalDate d = LocalDate.parse(test.group(0), cacherProperties.formatter);
+                            // collect all files that matches the build date pattern
+                            nightlyBuildArtifacts.put(file, d);
+                        }
+                    });
+
+            List<LocalDate> foundDates = new ArrayList<>(new HashSet<>(nightlyBuildArtifacts.values()));
+            LocalDate mostRecentNightlyBuild = Collections.max(foundDates);
```

Comment:
Actually we can do it simple :) `map.values()` return collection, we can pass it as an entry to Collections.max
```suggestion
            LocalDate mostRecentNightlyBuild = Collections.max(nightlyBuildArtifacts.values());
```

Reply:
good catch, updated.